src/pages/projectSettings/tabs/ViewsAndFiltersTab/ViewsAndFiltersTab.tsx

```diff
@@ -30,6 +30,23 @@ export const ViewsAndFiltersTab: React.VFC<TabProps> = ({ projectData }) => {
       onChange={onChange}
       schema={schema}
       uiSchema={uiSchema}
+      validate={validate as any}
     />
   );
 };
+
+/* Display an error and prevent saving if a user enters a Parsley filter expression that already appears in the project. */
+const validate = ((formData, errors) => {
+  const duplicateIndices = formData.parsleyFilters
+    .map((p) => p.expression)
+    .map((exp, i, arr) => exp !== "" && arr.lastIndexOf(exp) !== i && i)
+    .filter((i) => formData.parsleyFilters[i]);
```

Comment:
Do you think it's worth pulling this duplicator function out into a SpruceForm util?